From 181b47e4a25dc2e98c76b6094e09acb650160acb Mon Sep 17 00:00:00 2001
Message-Id: <181b47e4a25dc2e98c76b6094e09acb650160acb.1354557217.git.minovotn@redhat.com>
From: Yonit Halperin <yhalperi@redhat.com>
Date: Thu, 29 Nov 2012 16:59:45 +0100
Subject: [PATCH] qxl: reload memslots after migration, when qxl is in
 UNDEFINED mode

RH-Author: Yonit Halperin <yhalperi@redhat.com>
Message-id: <1354208385-11315-1-git-send-email-yhalperi@redhat.com>
Patchwork-id: 44518
O-Subject: [PATCH RHEL6.4] qxl: reload memslots after migration, when qxl is in UNDEFINED mode
Bugzilla: 874574
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Alon Levy <alevy@redhat.com>

The devram memslot stays active when qxl enters UNDEFINED mode (i.e, no
primary surface). If migration has occurred while the device is in
UNDEFINED stae, the memslots have to be reloaded at the destination.

Fixes rhbz#874574

Signed-off-by: Yonit Halperin <yhalperi@redhat.com>
---
 hw/qxl.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/qxl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/qxl.c b/hw/qxl.c
index 074b0eb..84b1398 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -2040,6 +2040,7 @@ static int qxl_post_load(void *opaque, int version)
 
     switch (newmode) {
     case QXL_MODE_UNDEFINED:
+        qxl_create_memslots(d);
         break;
     case QXL_MODE_VGA:
         qxl_create_memslots(d);
-- 
1.7.11.7

